woocommerce/woocommerce-android
# [WOO POS]M2: Filter Simple Products for POS Display

Closes: #12103 


### Description

This PR introduces a filtering mechanism for the product list in POS mode. The filter ensures that only relevant products are displayed based on specific criteria, enhancing the user experience in the POS environment.

Criteria Implemented:

1. The product must be published.
2. The product must have a price.
3. The product must be non-virtual.
4. The product must be non-downloadable.
5. This filtering approach ensures that only appropriate products are shown in the POS, aligning with the POC requirements.

### Testing information

1. Add different products that are not published, does not have a price, virtual, downloadable
2. Open WooCommerce app and ensure you are on the same store as the one you just added your products for in step 1
3. Navigate to the settings and then to the POS
4. Ensure you don't see any of the products that you created in the step 1
5. Ensure all other physical simple products are displayed as expected.

### Images/gif


https://github.com/user-attachments/assets/09bf932f-54be-4234-aaca-4f1cda515495
